plugin/helpers_bots.go

```diff
@@ -4,12 +4,81 @@
 package plugin
 
 import (
+	"io/ioutil"
+	"path/filepath"
+
 	"github.com/mattermost/mattermost-server/model"
 	"github.com/mattermost/mattermost-server/utils"
 	"github.com/pkg/errors"
 )
 
-func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error) {
+type EnsureBotOptions struct {
+	ProfileImagePath string
+	IconImagePath    string
+}
+
+type EnsureBotOption func(*EnsureBotOptions)
+
+func ProfileImagePath(path string) EnsureBotOption {
+	return func(args *EnsureBotOptions) {
+		args.ProfileImagePath = path
+	}
+}
+
+func IconImagePath(path string) EnsureBotOption {
+	return func(args *EnsureBotOptions) {
+		args.IconImagePath = path
+	}
+}
+
+func (p *HelpersImpl) readFile(path string) ([]byte, error) {
+	bundlePath, err := p.API.GetBundlePath()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get bundle path")
+	}
+
+	imageBytes, err := ioutil.ReadFile(filepath.Join(bundlePath, path))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read image")
+	}
+	return imageBytes, nil
+}
+
+func (p *HelpersImpl) setBotIconAndProfileImage(botId string, profileImagePath string, iconImagePath string) error {
+	if !(profileImagePath == "") {
+		imageBytes, err := p.readFile(profileImagePath)
+		if err != nil {
+			return errors.Wrap(err, "Failed to read profile image")
+		}
+		setProfileErr := p.API.SetProfileImage(botId, imageBytes)
+		if setProfileErr != nil {
+			return errors.Wrap(err, "Failed to set profile image")
+		}
+	}
+	if !(iconImagePath == "") {
```

Comment:
same comment as above, i think it'll be easier if this was `iconImagePath != ""`